### ISSUE TITLE
How to get the fc7 layer

### ISSUE THREAD

**Issue by orchardchang** [facebookresearch/pythia#111]
## ❓ Questions and Help
After reading paper _Pythia v0.1: the Winning Entry to the VQA challenge 2018_, I have some questions:

- How to fine tune the fc7 layer when I get the fc6 features? 

- Do you mean learning rate is 0.1 times when training pythia model with fine-tuned features?

- How the grid-features and object-features concatenated when training? Could you provide the code in Pythia?

**Comment by apsdehal** [comment]
For your questions, you should have a look at Pythia model at https://github.com/facebookresearch/pythia/blob/master/pythia/models/pythia.py and description of new model in TextVQA paper at https://arxiv.org/abs/1904.08920

Replying inline:
> How to fine tune the fc7 layer when I get the fc6 features?
The image encoder layer inside the pythia model is the one you are looking for. We take the weights of the fc7 layer from faster-rcnn and finetune them in our model

> Do you mean learning rate is 0.1 times when training pythia model with fine-tuned features?

Learning rate for the fc7 layer that we copied from the faster-rcnn has 0.1 lr compared to rest of the modules

> How the grid-features and object-features concatenated when training? Could you provide the code in Pythia?
They are concatenated in the end. https://github.com/facebookresearch/pythia/blob/master/pythia/models/pythia.py#L266

**Comment by orchardchang** [comment]
Thank you for your help. 
> The image encoder layer inside the pythia model is the one you are looking for. We take the weights of the fc7 layer from faster-rcnn and finetune them in our model.

How could I get the weights of the fc7 layer or design a fine-tuning layer from faster-rcnn if I prepare to fine tune the pre-trained features by myself? In fact I want to know the procedure of fine-tuning in faster rcnn.

**Comment by apsdehal** [comment]
Weights for fc7 layer are present at http://dl.fbaipublicfiles.com/pythia/data/detectron_weights.tar.gz.
We take fc6 layer's features and fine-tune the fc7 layer.

**Comment by Zzmonica** [comment]
Do I need to download the weights of `detectron_fix_100.tar.gz` 
It seems a little big to download.

**Comment by apsdehal** [comment]
You need this http://dl.fbaipublicfiles.com/pythia/data/detectron_weights.tar.gz.
